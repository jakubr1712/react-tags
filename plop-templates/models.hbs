export interface I{{properCase name}}ComponentProps {}