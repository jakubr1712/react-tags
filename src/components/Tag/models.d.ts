export interface ITagProps {
  id: number;
  name: string;
  onRemove: (id: number) => void;
}
