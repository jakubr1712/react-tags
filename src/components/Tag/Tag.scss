.tag {
  border: 1px solid #e0e0e0;
  border-radius: 10px;
  padding: 0px 8px;
  color: black;
  font-weight: 700;
  display: flex;
  align-items: center;

  &__remove-button {
    background: none;
    border: none;
    margin-left: 8px;
    cursor: pointer;
    color: #999;
    font-size: 24px;
    font-weight: 100;
  }
}
