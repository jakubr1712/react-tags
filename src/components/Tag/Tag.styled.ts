import styled from "styled-components";

export const TagWrapper = styled.div`
  border: 1px solid #e0e0e0;
  border-radius: 10px;
  padding: 0px 8px;
  color: black;
  font-weight: 700;
  display: flex;
  align-items: center;
`;

export const RemoveButton = styled.button`
  background: none;
  border: none;
  margin-left: 8px;
  cursor: pointer;
  color: #999;
  font-size: 24px;
  font-weight: 100;
`;
