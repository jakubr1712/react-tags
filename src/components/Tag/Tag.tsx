import React from "react";
import { TagWrapper, RemoveButton } from "./Tag.styled";

interface TagProps {
  id: number;
  name: string;
  onRemove: (id: number) => void;
}

const Tag: React.FC<TagProps> = ({ id, name, onRemove }) => (
  <TagWrapper>
    {name}
    <RemoveButton onClick={() => onRemove(id)}>&times;</RemoveButton>
  </TagWrapper>
);

export default Tag;
