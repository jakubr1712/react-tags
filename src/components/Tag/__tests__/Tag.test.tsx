import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Tag from "../Tag";
import { ITagProps } from "../models.d";

describe("Tag component", () => {
  const mockOnRemove = jest.fn();

  const defaultProps: ITagProps = {
    id: 1,
    name: "Sample Tag",
    onRemove: mockOnRemove,
  };

  beforeEach(() => {
    mockOnRemove.mockClear();
  });

  it("renders properly and matches snapshot", () => {
    const { container } = render(<Tag {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("displays the correct tag name", () => {
    render(<Tag {...defaultProps} />);
    expect(screen.getByText("Sample Tag")).toBeInTheDocument();
  });

  it("calls onRemove with correct id when remove button is clicked", () => {
    render(<Tag {...defaultProps} />);
    fireEvent.click(screen.getByText("Ã—"));
    expect(mockOnRemove).toHaveBeenCalledWith(1);
  });
});
