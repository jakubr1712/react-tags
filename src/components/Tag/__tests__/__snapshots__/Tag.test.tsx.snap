// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component renders properly and matches snapshot 1`] = `
<div>
  <div
    class="tag"
  >
    Sample Tag
    <button
      class="tag__remove-button"
    >
      Ã—
    </button>
  </div>
</div>
`;
